### PR TITLE
Report display sizes in physical pixels on MacOS

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/142629

`Display.size`'s [documentation](https://main-api.flutter.dev/flutter/dart-ui/Display/size.html) says (emphasize mine):

> The **physical** size of this display.

But we have actually been reporting the size in logical pixels - up until now!

### CODE REVIEW

**Comment by matanlurey** [comment]
Just an FYI: Both Chris and Dan are out the remainder of this week.

File: shell/platform/darwin/macos/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -834,13 +834,14 @@ - (void)updateDisplayConfig {
...
     FlutterEngineDisplay display;
     display.struct_size = sizeof(display);
     display.display_id = displayID;
     display.single_display = false;
-    display.width = static_cast<size_t>(screen.frame.size.width);
-    display.height = static_cast<size_t>(screen.frame.size.height);
-    display.device_pixel_ratio = screen.backingScaleFactor;
+    display.width = static_cast<size_t>(screen.frame.size.width) * devicePixelRatio;
+    display.height = static_cast<size_t>(screen.frame.size.height) * devicePixelRatio;
+    display.device_pixel_ratio = devicePixelRatio;
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Source/FlutterEngine.mm:844]: For testing this, I think we could probably extract the fetching of screens out as a method in the engine_internal.h header. In the test, we use a subclass that overrides the method providing the screens, call updateDisplayConfig, and verify we got a NotifyDisplayUpdate call to the embedder API with the expected size/resolutions.

Comment (goderbauer) [reply shell/platform/darwin/macos/framework/Source/FlutterEngine.mm:848]: Hm, the test fails on line 1183 indicating that the embedder's `NotifyDisplayUpdate` is never called. I traced it down to line 832 above: _engine is null at this point and we hit the early return. Any idea what my test is missing to get FlutterEngine into the right state where _engine is not null?

Comment (goderbauer) [reply shell/platform/darwin/macos/framework/Source/FlutterEngine.mm:848]: Oh, I think I am missing a call to `[engine runWithEntrypoint:@"main"]`. Let me try that.

Comment (cbracken) [reply shell/platform/darwin/macos/framework/Source/FlutterEngine.mm:848]: Awesome looks great!

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Pull request merged**